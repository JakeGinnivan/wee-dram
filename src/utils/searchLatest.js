export const searchLatest = (haveResults, performSearch) => {
    let currentSearchTerm

    return {
        searchTermChanged: (searchTerm) => {
            currentSearchTerm = searchTerm

            setTimeout(() => {
                if (currentSearchTerm === searchTerm) {
                    performSearch(searchTerm).then(r => {
                        if (currentSearchTerm === searchTerm) {
                            haveResults(r)
                        }
                    }).catch(err => console.log(err))
                }
            }, 500)
        }
    }
}