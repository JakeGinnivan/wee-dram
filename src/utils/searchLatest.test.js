import { searchLatest } from './searchLatest'

it('it searches after timeout', (done) => {
    const searchApi = searchLatest(results => {
        expect(results).toMatchSnapshot()
        done()
    }, () => {
        return Promise.resolve([{ id: 1, name: 'Glenmorangie Signet', region: '' }])
    })

    searchApi.searchTermChanged('glen')
});
