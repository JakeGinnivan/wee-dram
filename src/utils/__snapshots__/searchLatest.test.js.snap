exports[`test it searches after timeout 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Glenmorangie Signet",
    "region": "",
  },
]
`;
